File: 2593-find-score-of-an-array-after-marking-all-elements/2593-find-score-of-an-array-after-marking-all-elements.py
class Solution:
    def findScore(self, nums: list[int]) -> int:
        n = len(nums)
        total_sum = 0
        r = 0
        
        while r < n:
            l = r
            while r + 1 < n and nums[r + 1] < nums[r]:
                r += 1
            for i in range(r, l - 1, -2):
                total_sum += nums[i]
            r += 2 
        
        return total_sum

File: 2593-find-score-of-an-array-after-marking-all-elements/2593-find-score-of-an-array-after-marking-all-elements.cpp
class Solution {
public:
    long long findScore(vector<int>& nums) {
   
        int n = nums.size();
        long long sum = 0;
        for(int l,r = 0 ; r < n;r+=2){
            l = r;
            while (r+1 < n && nums[r+1] < nums[r]) r++;
            for(int i = r ; i >= l ; i-=2)sum+=nums[i];
        }
        return sum;
    }
};